addons/delivery/models/sale_order.py

```diff
@@ -112,6 +112,14 @@ def _create_delivery_line(self, carrier, price_unit):
         if self.order_line:
             values['sequence'] = self.order_line[-1].sequence + 1
         sol = SaleOrderLine.sudo().create(values)
+
+        # Adapt delivery line based on invoice policy
+        if carrier.invoice_policy == 'real':
+            sol.price_unit = 0
+            if sol.currency_id.position == 'before':
+                sol.name += ' (Estimated Cost: ' + sol.currency_id.symbol + str(price_unit) + ')'
+            else:
+                sol.name += ' (Estimated Cost: ' + str(price_unit) + sol.currency_id.symbol + ')'
```

Comment:
you can prepare the data in `values` before the call to create, no?
you should try to translate 'estimated cost'
you should format the price_unit and the currency_symbol similar to what's done [here](https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/ir_qweb_fields.py#L424)